resources/build_essential.rb

```diff
@@ -0,0 +1,98 @@
+#
+# Cookbook Name:: build-essential
+# resource:: build_essential
+#
+# Copyright 2008-2016, Chef Software, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+provides :build_essential
+resource_name :build_essential
+
+property :compile_time, [true, false], default: false
+
+action :install do
+  case node['platform_family']
+  when 'debian'
+    package %w( autoconf binutils-doc bison build-essential flex gettext ncurses-dev )
+  when 'fedora', 'rhel'
+    package %w( autoconf bison flex gcc gcc-c++ gettext kernel-devel make m4 ncurses-devel patch )
+
+    # Ensure GCC 4 is available on older pre-6 EL
+    package %w( gcc44 gcc44-c++ ) if node['platform_version'].to_i < 6
```

Comment:
That's the pattern I've been taking in other places, but I think there's less need / desire to do overrides here.  This is a pretty simple cookbook and if you need something funky you're probably better off just writing your own 1 line package resource.